prysm/io.py

```diff
@@ -1375,3 +1375,94 @@ def read_sigfit_rigidbody(file):
             'dR': dR
         }
     return out
+
+def read_codev_int(filename):
+    """Read the content of a CodeV INT (.int/.cvg) file.
+    
+    Parameters
+    ----------
+    file : path_like
+        path to a file
+    
+    Returns
+    -------
+    'dict'
+        dictionary with keys: phase, intensity, meta
+
+    """
+    with open(filename, 'rt') as f:
+        # skip headerlines starting with "!"
+        while True:
+            title = f.readline()
+            if not title.lstrip().startswith('!'):
+                break
+        # read parameter line
+        param = f.readline().split()
+        # parse parameters
+        if param[0] == 'GRD':
+            num_cols = int(param[1])
+            if param[2] == 'R':
+                #radial grid data
+                #not implemented
+                raise ValueError('Radial Grid Data INT-File type not supported.')
+            else:
+                num_rows = int(param[2])
+            datatype = param[3]    #SUR | WFR | FIL
+            #need to test now because different orders are allow
+            p=4
+            while p < len(param):
+                if param[p].upper() == 'WVL':
+                    wavelength = float(param[p+1])
+                if param[p].upper() == 'NMB':
+                    pass
+                if param[p].upper() == 'SSZ':
+                    scale_size = float(param[p+1])
+                if param[p].upper() == 'NDA':
+                    nda = int(param[p+1])
+                if param[p].upper() == 'XSC':
+                    x_scale = float(param[p+1])
+                p += 1    
+            #read grid data
+            phase=np.array([], dtype=config.precision)
+            while True:
+                line = f.readline()
+                if not line:
+                    break
+                phase = np.append(phase,[int(i) for i in line.split()])
+            
+            if not len(phase) == num_cols*num_rows:
+                raise ValueError('Error during INT-file import. Length of list does not match row*cols')
```

Comment:
prysm uses the convention of letting shape mismatch and similar low-level errors bubble up.  The length mismatch ValueError here provides no additional information vs letting numpy crash.  Actually, the numpy error contains the shapes that mismatched which is more information!

Reply:
INT Files have in the header a row col info. The data must match otherwise it is an error in the INT file. Therefore I would like to keep the info for the user, that there is an error in the INT File.